src/main.c

```diff
@@ -39,9 +40,9 @@ void train(neural_network *nn, dataset *train_dataset, dataset *test_dataset, fl
         dataset *batch_dataset = get_random_dataset_sample(train_dataset, batch_size);
         for (size_t i = 0; i < batch_dataset->length; i++) {
             data *data = batch_dataset->datas[i];
-            rotate_data(data, IMAGE_SIZE, IMAGE_SIZE, random_float(-5.0f, 5.0f));
-            scale_data(data, IMAGE_SIZE, IMAGE_SIZE, random_float(0.9f, 1.1f));
-            offset_data(data, IMAGE_SIZE, IMAGE_SIZE, random_float(-3.0f, 3.0f), random_float(-3.0f, 3.0f));
+            rotate_data(data, IMAGE_SIZE, IMAGE_SIZE, randf(&randc, -5.0f, 5.0f));
+            scale_data(data, IMAGE_SIZE, IMAGE_SIZE, randf(&randc, 0.9f, 1.1f));
+            offset_data(data, IMAGE_SIZE, IMAGE_SIZE, randf(&randc, -3.0f, 3.0f), randf(&randc, -3.0f, 3.0f));
```

Comment:
The range and offset is wrong it should be
```suggestion
            rotate_data(data, IMAGE_SIZE, IMAGE_SIZE, randf(&randc, 10.0f, -5.0f));
            scale_data(data, IMAGE_SIZE, IMAGE_SIZE, randf(&randc, 1.2f, -0.1f));
            offset_data(data, IMAGE_SIZE, IMAGE_SIZE, randf(&randc, 6.0f, -3.0f), randf(&randc, 6.0f, -3.0f));
```